pilot/pkg/model/service.go

```diff
@@ -84,6 +86,29 @@ type Service struct {
 	// or use the passthrough model (i.e. proxy will forward the traffic to the network endpoint requested
 	// by the caller)
 	Resolution Resolution
+
+	// Name is a unique immutable identifier in a namespace
+	Name string `json:"name,omitempty"`
```

Comment:
I'm assuming that `Name` refers to the name of the config `ServiceEntry` for this service? If that's the case, since a single `ServiceEntry` supports multiple hosts, would there potentially be multiple `model.Service`s with the same `Name`?

Reply:
Should we use [ConfigMeta](https://github.com/istio/istio/blob/7bba996125b321cbe694e057e1e8f0dfc4386593/pilot/pkg/model/config.go#L43) for this stuff?

Reply:
ConfigMeta is in a different package - I cut&pasted from there. It's not the host name - but the name of the K8S resource that created the object, so we can display meaningful messages ("your config foo in namespace bar is bad, please delete")

Reply:
We cannot. Because we are not processing service updates in order. We also have no guarantees on order of addition or removal. If you are doing this based on Karl’s suggestion it won’t work unless you rewrite all of networking logic to handle the service array as a versioned array. If you try to pre validate inside k8s adapter, it might work but for that you don't need to add k8s stuff into the service model.